src/main/scala/encry/modifiers/state/box/EncryBaseBox.scala

```diff
@@ -20,16 +20,14 @@ trait EncryBaseBox extends Box[EncryProposition] with PConvertible {
 
   override val tpe: Types.Product = Types.EncryBox
 
-  override def asVal: PValue = PValue(convert, tpe)
+  override def asVal: PValue = PValue(asPrism, tpe)
 
-  def convert: PObject = {
-    val fields = Map(
+  def asPrism: PObject =
```

Comment:
A think that you can move some common logic such as adding fields like `contractHash`, `typeId` etc into EncryBaseBox